### PR TITLE
1935 owl22casl sentence translation

### PR DESCRIPTION
**Pull request by mcodescu**
Should improve #1935.

### CODE REVIEW

File: OWL2/OWL22CASL.hs [hunk 1]
```diff
@@ -414,37 +415,37 @@ mapComObjectPropsList cSig mol props a b = do
...
 mapFacet :: CASLSign -> Int -> (ConstrainingFacet, RestrictionValue)
-    -> Result (CASLFORMULA, CASLSign)
+    -> Result (CASLFORMULA, [CASLSign])
 mapFacet sig var (f, r) = do
     con <- mapLiteral r
     let (fp, cf) = mkFacetPred con f var
-    return (fp, uniteCASLSign sig $ (emptySign ())
-            {predMap = MapSet.fromList [(cf, [dataPred])]})
+    return (fp, [(emptySign ())
+            {predMap = MapSet.fromList [(cf, [dataPred])]}])
```
Comment (tillmo) [inline OWL2/OWL22CASL.hs:449]: shouldn't `sig` appear in the resulting signature list?
why s the empty sig needed in the list?

File: OWL2/OWL22CASL.hs [hunk 2]
```diff
@@ -495,77 +496,77 @@ mapCard b cSig ct n prop d var = do
         exactLst' = mkImpl (conjunct $ oPropsE ++ fOut) $ disjunct dlstM
         senAux = conjunct [minLst, mkForall qVarsE exactLst']
         exactLst =  if null qVars then senAux else mkExist qVars senAux
-        ts = uniteL $ [cSig] ++ s ++ s' ++ s''
+        ts = concat $ s ++ s' ++ s''
```
Comment (tillmo) [inline OWL2/OWL22CASL.hs:499]: why can `csig` be omitted here (and below)?

**Review by tillmo — COMMENTED** [review]
My tests show that everything works fine. However, I do not understand the code at some places.

File: OWL2/OWL22CASL.hs (hunk 1 above)
Comment (mcodescu) [reply OWL2/OWL22CASL.hs:449]: no, `sig` should actually be omitted from all these functions. I will do this. 
I kept the old formatting and it's hard to read, it should be the signature that you get by inserting (cf, dataPred) in the predMap of the empty signature.

File: OWL2/OWL22CASL.hs (hunk 2 above)
Comment (mcodescu) [reply OWL2/OWL22CASL.hs:499]: csig is not needed, it was passed around just so we can return a signature when no change happens, and to be extended with the new symbols. I will remove it throughout the code.

Comment (mcodescu) [reply OWL2/OWL22CASL.hs:499]: Also, note that csig is added at the union in mapTheory anyways, and you don't want to have it there more than once.

**Review by tillmo — APPROVED** [review]

**Pull request merged**